services/secret-service/doc/openapi.yaml

```diff
@@ -299,22 +339,87 @@ components:
         - clientSecret
         - refreshTokenUrl
       properties:
-        name:
-          type: string
         clientId:
           type: string
         clientSecret:
           type: string
-        refreshTokenUrl:
+        redirectUri:
           type: string
+        endpoint:
```

Comment:
What is the purpose of this object?

Reply:
We need authentication and token refreshing endpoints. These are grouped in this object.

Reply:
The names of this object don't match the names in the OAuth2 spec. What are these properties for:

* userinfo
* revocation
* endSession

And what is the match in the OAuth2 spec?

Reply:
I would propose to stick to a flat structure and give url properties speaking names `fooUrl` such as `refreshTokenUrl`, `revokeTokenUrl`, etc.

Reply:
Section `3. Protocol endpoints` of [OAuth2 spec ](https://tools.ietf.org/html/rfc6749#page-18) mentions `Authorization endpoint` and `Token endpoint`, albeit the implementation details and naming of these endpoints is left to the protocol implementers. We can make these properties more speaking, e.g. `authUrl` and `tokenUrl`, but according to the spec, [token endpoint](https://tools.ietf.org/html/rfc6749#section-3.2) is responsible - among others - for refresh tokens. 
```
The token endpoint is used by the client to obtain an access token by
   presenting its authorization grant or refresh token.  The token
   endpoint is used with every authorization grant except for the
   implicit grant type (since an access token is issued directly).
```
Still, we think it makes sense to group authorization and token endpoints in one object, instead of a flat structure.
`userinfo` is meant to fetch user data. Let's say the user creates two Google/Microsoft/Atlassian secrets through the authClient and the secret contains only the tokens. Accessing userinfo would fetch username/email, so that the these can be displayed to a user through UI.
Similar to `mappings` we could remove these properties from the spec and add them later, if needed. Our implementation of secrets-service will still support these features.

Reply:
Ok